processing/src/main/java/org/apache/druid/indexer/Checks.java

```diff
@@ -36,12 +37,18 @@ public static <T> Property<T> checkOneNotNullOrEmpty(List<Property<T>> propertie
         if (nonNullProperty == null) {
           nonNullProperty = property;
         } else {
-          throw new IAE("At most one of %s must be present", properties);
+          throw new IAE(
+              "At most one of %s must be present",
+              properties.stream().map(Property::getName).collect(Collectors.toList())
```

Comment:
Maybe String.join() the items using ' or ' .

Reply:
IMO, with the interpolation it's clear, it'll look like:

```
At most one of properties[[a, b]] must be present
```